src/index.js

```diff
@@ -184,13 +184,7 @@ class TerserPlugin {
   }
 
   static hasAsset(commentFilename, assets) {
-    const assetFilenames = Object.keys(assets).map((assetFilename) =>
-      TerserPlugin.removeQueryString(assetFilename)
-    );
-
-    return assetFilenames.includes(
-      TerserPlugin.removeQueryString(commentFilename)
-    );
+    return assets.has(TerserPlugin.removeQueryString(commentFilename));
```

Comment:
Why it was changed, refactor? Please avoid not relevant changes

Reply:
I'm sorry that I didn't make clear why this change is necessary. I extracted this change into a separate commit 35f310c0a22da98bc459a7d8729ff1cb2879c2bc and explain why it's necessary there. Let me know if it's still not clear.

Reply:
It is expected, two difference tasks should not generate same files